### PR TITLE
WIP: Add form styles

### PR DESCRIPTION
**Pull request by KennethAshley**
##### Description
We want to have consistent form styles across Gitcoin

##### Checklist

- [ ] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)
UI changes

##### Refers/Fixes
Refs: #498

##### Todo
- [x] Create Work form
- [x] Submit Work form
- [x] Kill Bounty
- [x] Sidebar search
- [x] Slack page
- [x] Hints
- [x] Email settings
- [x] whitepaper

Anymore that i missed?

### CODE REVIEW

**Comment by KennethAshley** [comment]
Fund Work
screencapture-localhost-8000-funding-new-1519887237112

**Comment by KennethAshley** [comment]
@PixelantDesign @owocki what other forms need to be updated?

**Comment by thelostone-mc** [comment]
@KennethAshley There is one in https://gitcoin.co/slack

**Comment by PixelantDesign** [comment]
Looking good @KennethAshley! I was hoping that all forms across the Gitcoin.co site would be updated as described at the top of the ticket.

This includes: 
- The form and all states
- font updates
- hint text styling

Let me know if you have any other questions! Thanks!

**Comment by KennethAshley** [comment]
screencapture-localhost-8000-funding-new-1520033403645
screencapture-localhost-8000-funding-fulfill-1520033435393
screencapture-localhost-8000-funding-kill-1520033470700
image
screencapture-localhost-8000-slack-1520033530567
screencapture-localhost-8000-kenneth-1520033627006
screencapture-localhost-8000-whitepaper-1520033679029

@PixelantDesign @thelostone-mc What do you guys think?

**Comment by KennethAshley** [comment]
There was alot of small little changes here and there. So i might have missed something. Please let me know if i did.

**Comment by PixelantDesign** [comment]
Looking good @KennethAshley 
Just a few more things...will you take a look at the following?

Line up form label and form field on all forms
screen shot 2018-03-02 at 8 20 00 pm

In the interested in box, could we change that to a drop down? Might just simplify things.
screen shot 2018-03-02 at 8 15 36 pm

Can we remove a few things and reorder to look like this?
screen shot 2018-03-02 at 8 27 28 pm

Can we make the corner radius of the drop downs match the radius of the input fields?
screen shot 2018-03-02 at 8 11 47 pm

Otherwise, I think we're close! Thank you!

**Comment by KennethAshley** [comment]
@PixelantDesign if i change the multiple to a dropdown i'm not sure how the user will select multiple. We might have to rethink that to be a group of checkboxes? Let me know what you think. A few solutions (in terms of difficulty) would be a group of checkboxes (low) styling the multiple select element (medium) using the already installed select2 plugin (medium-hard).

**Comment by KennethAshley** [comment]
screencapture-localhost-8000-slack-1520059641030
screencapture-localhost-8000-funding-new-1520059675629

**Comment by thelostone-mc** [comment]
@KennethAshley this looks solid :D

**Comment by thelostone-mc** [comment]
@KennethAshley Quick question : 
I'm working on the explorer page -> and I'm aware once your PR gets merged,  I'm gonna have a shit ton of conflicts to resolve 

I'm gonna try and avoid rewriting css -> so help me out here

To use your input_field css
` <input class= "form__input" ... />`

And for buttons css
`<a class="button button--primary"...></a>`

**Comment by KennethAshley** [comment]
@thelostone-mc yea that's how you use them. I was going with BEM style class names.

**Comment by PixelantDesign** [comment]
@kennethashley thanks for the LOE, that was helpful context to have. For multi select on the whitepaper page, lets replace with checkboxes for now. 

Thank you! Looking so good!

**Comment by olafghanizadeh** [comment]
It would be great to try to use bootstrap helper classes going forward and keep it consistent to avoid writing custom and hacky CSS.

**Comment by PixelantDesign** [comment]
I agree, thats sounds like a good approach.

**Comment by KennethAshley** [comment]
@olafghanizadeh that's the issue with using a framework like bootstrap and having custom tailored designs. You'll have to write hacky css to get it to look how you want. So instead I left off the bootstrap classes and styled from the ground up.

**Comment by olafghanizadeh** [comment]
No, that is not correct. If you use Bootstrap correctly, you only overwrite what you need to overwrite and leave the rest as is to secure cross-browser compatibility and responsiveness, which is arguably the main advantage of using a framework like Bootstrap.

Ideally you use Bootstrap, and then you overwrite the values you need to overwrite, if you are using SCSS as well this will be compiled properly on build.

**Comment by thelostone-mc** [comment]
@PixelantDesign could we leave the checkbox bit aside for now ? 
I've already gotten the css and stuff written in my PR

**Comment by KennethAshley** [comment]
@olafghanizadeh yea but that's just it, you have to overwrite styles to get it to work with a design. What if we wanted an element that isn't within the bootstrap framework? We'd have to write it from scratch. 

Bootstrap is a great tool to bootstrap a site or build an mvp, especially if you don't have a great amount of time. But for custom work it could be more of a pain then a helper. 

You bring out some great points though!

**Comment by olafghanizadeh** [comment]
If used properly, you can easily create your own components with Bootstrap without having to write a lot of custom CSS, and it will still include media-queries for responsiveness and any cross-browser quirk fixes. However, this requires you to use a preprocessor like SCSS to compile your production CSS file. 

For instance, if I wanted to create a default checkbox-styling for gitcoin I could do:

```

.gitcoin-form-check {
    @extend .form-check-input;
    border-color: $gitcoin-primary;
   /*
FURTHER CUSTOMIZATION
*/
}
```

After compiling I could reuse this `.gitcoin-form-check` on all checkboxes throughout the page.

In my opinion the advantages of bootstrap are:
1) Heavily vetted and tested for responsiveness and cross-browser conflicts
2) Quick scaffolding.

**Comment by KennethAshley** [comment]
@olafghanizadeh I agree with your advantages of using bootstrap for sure. I have used a frameworks when there were a few devs and we needed to focus on functionality over looks.

If we aren't using a preprocessor like in Gitcoin's case then we have to overwrite the styles to get the styles correct and that can be more work then just building the styles from scratch. 

`@extend` needs to be used carefully. It doesn't copy styles from one selector to the other it groups selectors. Which for a small site this might be fine but for a large size this can create a of complexity in the long run.

Now a days you can include `reset.css` and `normalize.css` which is a great foundation for you to build stuff from scratch and not have to worry about cross-browser conflicts.

**Comment by KennethAshley** [comment]
@mbeacom what do you think about this PR? @PixelantDesign has green-lighted it.

File: app/retail/templates/shared/head.html [hunk 1]
```diff
@@ -32,6 +32,15 @@
...
     <link rel="stylesheet" href="{% static "v2/css/carousel.css" %}" />
 
+    <!-- Forms -->
+    <link rel="stylesheet" href="{% static "v2/css/forms/button.css" %}" />
+    <link rel="stylesheet" href="{% static "v2/css/forms/checkbox.css" %}" />
+    <link rel="stylesheet" href="{% static "v2/css/forms/form.css" %}" />
+    <link rel="stylesheet" href="{% static "v2/css/forms/input.css" %}" />
+    <link rel="stylesheet" href="{% static "v2/css/forms/label.css" %}" />
+    <link rel="stylesheet" href="{% static "v2/css/forms/radio.css" %}" />
+    <link rel="stylesheet" href="{% static "v2/css/forms/select.css" %}" />
```
Comment (mbeacom) [inline app/retail/templates/shared/head.html]: 😢

**Comment by mbeacom** [comment]
@KennethAshley One thing we're noticing on staging is that the 
screenshot 2018-03-05 14 35 05
for dropdown elements isn't clickable.  Would you mind resolving this in a separate branch off of master?

Thanks for the contribution! Great work!

**Pull request merged**